### ISSUE TITLE
Should logger log in process.env.NODE_ENV === test?

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#102]
Noticed it with https://github.com/oliviertassinari/material-ui/commit/9a3c143420a4e0bd6d1d5f010d37b43a62280b77.

**Comment by dtassone** [comment]
I think it can log from the info level. So it's less verbose and we can understand a bit more where it fails

**Comment by oliviertassinari** [comment]
I was looking that the console on the storybook side, the debug and info messages hide the other important warnings and errors.

What about the following plan? 

- We don't log debug nor info by default, nor in people's code, nor in our test, nor in the storybook. 
- We allow developers to opt-in for debut and info.

**Comment by dtassone** [comment]
How about we set the default level to warning? 
If a user use the grid and don't see the warning or errors of the component, it's not great.

How do you want to opt-in?

**Comment by oliviertassinari** [comment]
> How about we set the default level to warning?
If a user use the grid and don't see the warning or errors of the component, it's not great.

I was under the assumption that we use `console.warn` and `console.error` direclty for these warnings, bypassing any abstraction.

> How do you want to opt-in?

I think that it depends on the use cases. Maybe?

- If it's for internal development, we could imagine an env variable or a localeStorage variable.
- If it's for our users, we could image a prop too.

**Comment by dtassone** [comment]
Users have the option to pass their own logger, in the case they use a database or something like seq or splunk...
The default logger used by the grid is console. So by default it does console.warn but it can be overwritten to work with a real logger.
They can also set the minimum level they want the grid to log.

>If it's for internal development, we could imagine an env variable or a localeStorage variable.

It already does that. If you add DEBUG in your local storage, it overrides the environment conf and start logging. 

What are we trying to achieve here?
Should we just change the default log level to warn?

**Comment by oliviertassinari** [comment]
> What are we trying to achieve here?
Should we just change the default log level to warn?

This would be great :) It would solve the flooding pain point and enough to close the issue. 

> Users have the option to pass their own logger, in the case they use a database or something like seq or splunk...

Interesting, would it be used for remote debugging? For this use case, I was expecting that developers would hook into the native console calls and record all the events.

**Comment by dtassone** [comment]
Well I have not come through the practice of overriding the console to get the component logs. Usually library or component logging get dropped and forgotten as we expect libraries to work...
I've used https://github.com/trentm/node-bunyan in the past which was pushing to our seq interface https://datalust.co/

**Comment by oliviertassinari** [comment]
I never had this case in the browser, so I can't tell. I also didn't benchmark what alternative UI libraries have for logging. Let's see and explore how that custom logger will turn out :). 

The closest I was in the past, was with Sentry, they wrap EVERYTHING, we would use the console logs recorded to debug exceptions. On node.js side, I have tried node bunyan but their decoder wasn't forwarding the right system signal for our docker instance cleanup to work correctly. We went with a custom implementation and an adapter for ClouldWatch on AWS.

What problem were you solving with Seq?
